From 5fac79b089bd7a58c4fca75db5df9672994aba2b Mon Sep 17 00:00:00 2001
From: Robert Ancell <robert.ancell@canonical.com>
Date: Tue, 9 Apr 2019 14:08:03 +1200
Subject: [PATCH] region: Fix input source options not being applied.

This was broken in a86cf1eca2 (3.32) when the input options dialog was converted
into a GtkPopover. Previously two GSettings objects were used, with one set to
'delay-apply' mode. The input source option didn't apply the changes.

The 'delay-apply' mode is from the original commit d3852fc831, however code was
simplified in b3199dd to only set one setting. This meant that the delay is no
longer necessary.

Fixes #440
---
 panels/region/cc-region-panel.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/panels/region/cc-region-panel.c b/panels/region/cc-region-panel.c
index 35859526d..664097da2 100644
--- a/panels/region/cc-region-panel.c
+++ b/panels/region/cc-region-panel.c
@@ -951,7 +951,6 @@ set_input_settings (CcRegionPanel *self)
         }
 
         g_settings_set_value (self->input_settings, KEY_INPUT_SOURCES, g_variant_builder_end (&builder));
-        g_settings_apply (self->input_settings);
 }
 
 static void set_localed_input (CcRegionPanel *self);
@@ -1204,7 +1203,6 @@ static void
 setup_input_section (CcRegionPanel *self)
 {
         self->input_settings = g_settings_new (GNOME_DESKTOP_INPUT_SOURCES_DIR);
-        g_settings_delay (self->input_settings);
 
         self->xkb_info = gnome_xkb_info_new ();
 
-- 
2.20.1

